session/session.go

```diff
@@ -540,6 +541,22 @@ func (s *session) CommitTxn(ctx context.Context) error {
 
 	var commitDetail *execdetails.CommitDetails
 	ctx = context.WithValue(ctx, execdetails.CommitDetailCtxKey, &commitDetail)
+	is := infoschema.GetInfoSchema(s)
+	txnScope := s.GetSessionVars().TxnScope
```

Comment:
`SessionVars.TxnScope` may change during the transaction. A `TxnScope` should be saved in `TransactionContext`.

Reply:
updated.